src/AIS.hpp

```diff
@@ -44,6 +49,23 @@ namespace nmea0183 {
          */
         uint32_t getDiscardedSentenceCount() const;
 
+        /**
+         * Applies position correction using the vessel reference position and the sensor
+         * offset
+         *
+         * @param position the AIS position to be corrected
+         * @param vessel_reference_position The position of the vessel’s reference point
+         * relative to the sensor
+         * @param utm_converter The UTM converter
+         *
+         * @return The corrected AIS position with updated latitude, longitude, and
+         * correction status
+         */
+        static ais_base::Position applyPositionCorrection(
+            ais_base::Position const& position,
+            base::Vector3d const& vessel_reference_position,
```

Comment:
this is a transform and should be named as our convention, my suggestion is:

```suggestion
            base::Vector3d const& vessel2ais_gnss_pos,
```

Reply:
I personally prefer `sensor2vessel_in_vessel_pos`